Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,40 @@
+# Ref: http://wiki.bash-hackers.org/howto/mutex
+
+HOMEBREW_LOCK_DIR="$HOMEBREW_LIBRARY/lock.d"
+HOMWBREW_LOCK_PID_FILE="$HOMEBREW_LOCK_DIR/PID"
+
+lock() {
```

Comment:
I don't agree; `ps` will show the process before a single line of code is executed.